horace_core/sqw/file_io/@horace_binfile_interface/reopen_to_write.m

```diff
@@ -75,19 +72,14 @@
 end
 obj.file_id_ = sqw_fopen(fname,permissions);
 if isempty(obj.file_closer_)
-    obj.file_closer_  = onCleanup(@()fclose(obj));
+    obj.file_closer_  = fcloser(obj.file_id_);
 end
 
 
-
 function obj=fclose_file(obj)
 if obj.file_id_>0
-    % obj.file_closer_ = [];  % This should close the file but not
-    % immediately but randomly at the moment, dependent on Matlab version,
-    % so we should avoid doing this unless sure that if works as expected.
-    % We assume that automatic file closer would close file, associated
-    % with the object regardless of file id, currently attached to the
-    % object
+    obj.file_closer_.delete();
+    obj.file_closer_ = []; 
```

Comment:
Having done some experimentation is it necessary to clear `file_closer_` like this? `delete` leaves a `handle to deleted object` and attempting methods on this gives the following error, which is much more clear and helpful than an `invalid function on double`:
```
Invalid or deleted object.

Error in TmpFileHandler/get.ref_count (line 73)
            rc = obj.ref_count_;
```

Reply:
it is different file closer here.

Reply:
If you set the value to `[]` you make an empty `double` rather than a `deleted handle` which makes any error messages of subsequent operations (if any, which there shouldn't be) more obscure

Reply:
file handle is different from normal handle. if it is not accociated with file, it just a valid number. This is why this code works and have to be done this way.

Reply:
What you said makes no sense to me. I'm talking about removing the line setting it to `[]` as you have already deleted the handle. That deletion destroys and disassociates the object and thus will still not work if subsequently called.